books/forms.py

```diff
@@ -24,41 +24,29 @@ def clean_author(self):
 
     def clean_cover(self):
         cover = self.cleaned_data.get("cover")
-        if cover:
-            name, extension = os.path.splitext(cover.name)
-            extension = extension.lower()
-            book = self.getBookIfExists()
-            if not book or (book.cover != cover):
+        if not cover:
+            cover = None
+        if cover is not None:
+            if self.has_changed() and "cover" in self.changed_data:
```

Comment:
in python-style code, you don't need to check for "is `None`" unless you expect a difference between a "falsey" value and **exactly** `None`.
So the above you could re-write as:

```python
cover = self.cleaned_data.get("cover")
if cover and "cover" in changed_data:
    # ....
```

I  maybe be wrong, but I don't think you need to check if `self.has_changed()`? `self.changed_data` would just be empty if there are no changes.

Reply:
An example of where you may want to distinguish between `None` and "falsey" would be something like a "nullable boolean" which can be either `False` (set to false) or `None` (not set at all). ex you could have a variable like:

`email_subscription = True|False|None`

where False means the user refused and None means the user hasn't set up a preference yet.

Similarly you could have a distinction between `0` and `None`, for example:

```python
def get_grade(assignment):
    if assignment.submitted:
        return assignment.grade
    return None  # assignment is not submitted so a grade does not apply
```

But in most cases it is safe (and more readable) to just do `if variable` or `if not variable`.